src/rapids_metadata/metadata.py

```diff
@@ -12,36 +12,54 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from dataclasses import dataclass, field
+from enum import Enum
+from os import PathLike
 from typing import Union
 
+from .rapids_version import get_rapids_version
 
-class Metadata:
-    def __init__(self, inherit: Union["Metadata", None]=None):
-        self._packages: set[str] = set()
-        self._non_alpha_spec_packages: set[str] = set()
-        self._non_cuda_suffixed_packages: set[str] = set()
+__all__ = ["PseudoRepository", "RAPIDSMetadata", "RAPIDSPackage", "RAPIDSRepository", "RAPIDSVersion"]
 
-        if inherit:
-            self._packages.update(inherit.packages)
-            self._non_alpha_spec_packages.update(inherit.non_alpha_spec_packages)
-            self._non_cuda_suffixed_packages.update(inherit.non_cuda_suffixed_packages)
 
-    @property
-    def packages(self) -> set[str]:
-        return self._packages
+class PseudoRepository(Enum):
+    NVIDIA = "nvidia"
 
-    @property
-    def non_alpha_spec_packages(self) -> set[str]:
-        return self._non_alpha_spec_packages
+    def __str__(self):
+        return f"_{self.value}"
+
+
+@dataclass
+class RAPIDSPackage:
+    has_alpha_spec: bool = field(default=True)
+    has_cuda_suffix: bool = field(default=True)
+
+
+@dataclass
+class RAPIDSRepository:
+    packages: dict[str, RAPIDSPackage] = field(default_factory=dict)
+
+
+@dataclass
+class RAPIDSVersion:
+    repositories: dict[Union[str, PseudoRepository], RAPIDSRepository] = field(default_factory=dict)
 
     @property
-    def non_cuda_suffixed_packages(self) -> set[str]:
-        return self._non_cuda_suffixed_packages
+    def all_packages(self):
```

Comment:
```suggestion
    def all_packages(self) -> set[RAPIDSRepository]:
```

Can you please add return type hints on all these methods? That'd made it easier to understand the flow of data, and will help type checkers to catch bugs.